scipy/stats/_mannwhitneyu.py

```diff
@@ -23,15 +23,32 @@ class _MWU:
     def __init__(self):
         '''Minimal initializer'''
         self._fmnks = -np.ones((1, 1, 1))
+        self._recursive = None
 
     def pmf(self, k, m, n):
-        '''Probability mass function'''
+        if (self._recursive is None and m <= 500 and n <= 500
+                or self._recursive):
```

Comment:
We can debate separately about the best cases to switch and whether we should have different logic for `'auto'` vs when the user explicitly chooses `'exact'`; let's make this PR about adding a simple, iterative implementation and save refinements to optimize later.